hikari/channels.py

```diff
@@ -1362,3 +1399,134 @@ class GuildStageChannel(GuildChannel):
+    """
+
+    last_pin_timestamp: typing.Optional[datetime.datetime] = attr.field(eq=False, hash=False, repr=False)
+    """The timestamp of the last-pinned message.
+
+    !!! note
+        This may be `builtins.None` in several cases; Discord does not document what
+        these cases are. Trust no one!
+    """
+
+    rate_limit_per_user: datetime.timedelta = attr.field(eq=False, hash=False, repr=False)
+    """The delay (in seconds) between a user can send a message to this channel.
+
+    If there is no rate limit, this will be 0 seconds.
+
+    !!! note
+        Any user that has permissions allowing `MANAGE_MESSAGES`,
+        `MANAGE_CHANNEL`, `ADMINISTRATOR` will not be limited. Likewise, bots
+        will not be affected by this rate limit.
+    """
+
+    approximate_message_count: int = attr.field(eq=False, hash=False, repr=True)
+    """Approximate number of messages in the thread channel.
+
+    !!! warning
+        This stops counting at 50.
+    """
+
+    approximate_member_count: int = attr.field(eq=False, hash=False, repr=True)
+    """Approximate count of members in the thread channel.
+
+    !!! warning
+        This stop counting at 50.
+    """
+
+    is_archived: bool = attr.field(eq=False, hash=False, repr=True)
+    """Whether the thread is archived."""
+
+    auto_archive_duration: datetime.timedelta = attr.field(eq=False, hash=False, repr=True)
+    """How long the thread will be left inactive before being automatically archived.
+
+    As of writing this may be one of 60, 1_440, 4_320 or 10_080 minutes.
+    """
+
+    # TODO: does this count when it was last unarchived?
+    last_archived_at: datetime.datetime = attr.field(eq=False, hash=False, repr=True)
+    """When the thread was last archived.
+
+    !!! note
+        If the thread has never been archived then this will be the thread's
+        creation date.
+    """
+
+    is_locked: bool = attr.field(eq=False, hash=False, repr=True)
+    """Whether the thread is locked.
+
+    When a thread is locked, only users with `MANAGE_THREADS` permission
+    can un-archive it.
+    """
+
+    member: typing.Optional[ThreadMember] = attr.field(eq=False, hash=False, repr=True)
+    """Thread member object for the current user, if they are in the thread.
+
+    !!! note
+        This is only returned by some endpoints
+    """  # TODO: WHICH ENDPOINTS?!?!?!?!?!??!?!?!??!?!?!?!?
+
+    owner_id: snowflakes.Snowflake = attr.field(eq=False, hash=False, repr=True)
+    """ID of the user who created this thread."""
+
+    parent_id: snowflakes.Snowflake = attr.field(eq=False, hash=False, repr=True)
+    """Id of this thread's textable parent channel."""
+
+
+class GuildNewsThread(GuildThreadChannel):
+    """Represents a guild news channel public thread."""
+
+    __slots__: typing.Sequence[str] = ()
+
+
+class GuildPublicThread(GuildThreadChannel):
+    """Represents a non-news guild channel public thread."""
+
+    __slots__: typing.Sequence[str] = ()
```

Comment:
These should just use `attr.define`. It allows us to later be able to use attr functions on them without having to check

Reply:
that'd be unnecessary eval calls (calling attr.defined is nowhere near free) and you should be checking this stuff (committing and pushing changes without tests is just bad)